// @adaptavant/aw-assets/dist/Button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Button(props) {
  var type = props.type,
      isDisabled = props.isDisabled,
      hasArrow = props.hasArrow,
      children = props.children,
      onClick = props.onClick,
      isMedium = props.isMedium;
  return /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: (0, _classnames.default)({
      'g-btn-primary': type === 'primary',
      'g-btn-secondary': type === 'secondary',
      'g-btn-tertiary': type === 'tertiary',
      'g-btn-ghost': type === 'ghost',
      'g-btn-danger': type === 'danger',
      'has-arrow': hasArrow,
      'btn-md': isMedium
    }),
    onClick: onClick,
    disabled: isDisabled
  }, children);
}

;
var _default = Button;
exports.default = _default;
Button.propTypes = {
  type: _propTypes.default.string,
  onClick: _propTypes.default.func,
  isDisabled: _propTypes.default.bool,
  hasArrow: _propTypes.default.bool
};
Button.defaultProps = {
  type: 'primary',
  onClick: undefined,
  isDisabled: false,
  hasArrow: false
};
