// @adaptavant/aw-assets/dist/ContentLoader/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ContentLoaders(props) {
  var width = props.width,
      _props$height = props.height,
      height = _props$height === void 0 ? width : _props$height,
      className = props.className,
      type = props.type;

  switch (type) {
    case 'sidebar_loader':
      return /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("figure", {
        className: "g-avatar"
      }, /*#__PURE__*/_react.default.createElement("span", {
        className: "g-content-loader"
      })), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h4", {
        className: "g-content-loader"
      }), /*#__PURE__*/_react.default.createElement("span", {
        className: "g-content-loader"
      }), /*#__PURE__*/_react.default.createElement("i", {
        className: "g-time-stamp g-content-loader"
      })));

    default:
      return /*#__PURE__*/_react.default.createElement("em", {
        style: {
          width: width,
          height: height
        },
        className: "g-content-loader ".concat(className)
      });
  }
}

var _default = ContentLoaders;
exports.default = _default;
ContentLoaders.propTypes = {
  width: _propTypes.default.string,
  height: _propTypes.default.string,
  className: _propTypes.default.string,
  type: _propTypes.default.string
};
ContentLoaders.defaultProps = {
  width: undefined,
  height: undefined,
  className: '',
  type: ''
};
