// @adaptavant/aw-assets/dist/Popover/Popover.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Popover(props) {
  var children = props.children,
      position = props.position;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)({
      'g-popover': true,
      'right-pos': position === 'right',
      'left-pos': position === 'left',
      'top-pos': position === 'top',
      'bottom-pos': position === 'bottom'
    })
  }, children);
}

var Label = function Label(props) {
  var children = props.children;
  return /*#__PURE__*/_react.default.createElement("span", {
    className: "label"
  }, children);
};

var List = function List(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement("div", {
    class: "g-popover__content has-arrow"
  }, /*#__PURE__*/_react.default.createElement("ul", null, children));
};

var ListItem = function ListItem(_ref2) {
  var children = _ref2.children,
      _ref2$className = _ref2.className,
      className = _ref2$className === void 0 ? '' : _ref2$className;
  return /*#__PURE__*/_react.default.createElement("li", {
    className: className
  }, children);
};

Popover.Label = Label;
Popover.List = List;
Popover.ListItem = ListItem;
var _default = Popover;
exports.default = _default;
