// @adaptavant/aw-assets/dist/Avatar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Avatar(props) {
  var isLarge = props.isLarge,
      photo = props.photo,
      alt = props.alt;
  var isPhoto = photo && photo.length > 0;
  return /*#__PURE__*/_react.default.createElement("figure", {
    className: (0, _classnames.default)({
      'g-avatar': !isLarge,
      'g-avatar-lg': isLarge
    })
  }, isPhoto ? /*#__PURE__*/_react.default.createElement("img", {
    src: photo,
    alt: alt
  }) : /*#__PURE__*/_react.default.createElement("figcaption", null, alt[0]));
}

var _default = Avatar;
exports.default = _default;
Avatar.propTypes = {
  isLarge: _propTypes.default.bool,
  photo: _propTypes.default.string,
  alt: _propTypes.default.string
};
Avatar.defaultProps = {
  isLarge: false,
  photo: '',
  alt: 'avatar-photo'
};
