// @adaptavant/aw-assets/dist/Modal/Modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Icon = _interopRequireDefault(require("../Icon/Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Modal(props) {
  var children = props.children;
  return /*#__PURE__*/_react.default.createElement("div", {
    class: "g-modal-wrapper alternate"
  }, children);
}

var Header = function Header(_ref) {
  var children = _ref.children,
      icon = _ref.icon,
      onClose = _ref.onClose;
  var hasIcon = icon && icon.length;
  return /*#__PURE__*/_react.default.createElement("header", {
    className: "g-modal-head"
  }, hasIcon ? /*#__PURE__*/_react.default.createElement(_Icon.default, {
    icon: icon,
    className: "g-modal-image"
  }) : null, children, onClose ? /*#__PURE__*/_react.default.createElement(_Icon.default, {
    icon: "g-cross",
    className: "g-icon",
    wrapperClassName: "g-modal-close",
    width: "20px",
    onClick: onClose
  }) : null);
};

var ModalBody = function ModalBody(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/_react.default.createElement("section", {
    className: "g-modal-content"
  }, children);
};

var ModalFooter = function ModalFooter(_ref3) {
  var children = _ref3.children;
  return /*#__PURE__*/_react.default.createElement("footer", {
    className: "g-modal-foot"
  }, children);
};

Modal.Header = Header;
Modal.Body = ModalBody;
Modal.Footer = ModalFooter;
var _default = Modal;
exports.default = _default;
