// @adaptavant/aw-assets/dist/Dropdownv2/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Avatar = _interopRequireDefault(require("../Avatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure undefined"); }

function Dropdown(props) {
  var children = props.children;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "g-select-wrap"
  }, children);
}

Dropdown.ToggleButton = function (_ref) {
  _objectDestructuringEmpty(_ref);

  return /*#__PURE__*/_react.default.createElement("button", {
    class: "g-select-btn"
  }, /*#__PURE__*/_react.default.createElement(_Avatar.default, {
    photo: "https://avatarfiles.alphacoders.com/694/69494.png"
  }), /*#__PURE__*/_react.default.createElement("svg", {
    class: "g-select-icon"
  }, /*#__PURE__*/_react.default.createElement("use", {
    href: "#g-clock"
  })), /*#__PURE__*/_react.default.createElement("span", {
    class: "g-select-label"
  }, "Label text"), /*#__PURE__*/_react.default.createElement("span", {
    class: "g-select-text"
  }, "Drop Down"), /*#__PURE__*/_react.default.createElement("i", {
    class: "g-select-arrow-icon"
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "16",
    height: "16",
    viewBox: "0 0 16 16",
    fill: "none"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M8 12.666C7.73333 12.666 7.48267 12.562 7.294 12.374L0.146667 5.22729C0.052 5.13262 0 5.00729 0 4.87329C0 4.73996 0.052 4.61396 0.146667 4.51996C0.241333 4.42529 0.366667 4.37329 0.5 4.37329C0.633333 4.37329 0.759333 4.42529 0.853333 4.51996L8 11.666L15.1467 4.51996C15.2413 4.42529 15.3667 4.37329 15.5 4.37329C15.6333 4.37329 15.7593 4.42529 15.8533 4.51996C15.948 4.61462 16 4.73996 16 4.87329C16 5.00662 15.948 5.13262 15.8533 5.22662L8.70667 12.3733C8.51867 12.562 8.26733 12.666 8 12.666Z",
    fill: "#666666"
  }))));
};

var _default = Dropdown;
exports.default = _default;
