// @adaptavant/aw-assets/dist/Chip/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = _interopRequireDefault(require("../Icon/Icon"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Chip(props) {
  var children = props.children,
      isEnabled = props.isEnabled,
      isSelected = props.isSelected,
      isDisabled = props.isDisabled,
      hasCloseIcon = props.hasCloseIcon,
      type = props.type,
      avatar = props.avatar,
      initial = props.initial;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)({
      'g-chip': true,
      'is-enabled': isEnabled,
      'is-selected': isSelected,
      'is-disabled': isDisabled,
      'has-initial': type === 'initial',
      'has-avatar': type === 'avatar'
    })
  }, type === 'initial' ? /*#__PURE__*/_react.default.createElement("figure", null, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    icon: "g-tick",
    width: "24px",
    wrapperClassName: "g-chip-selected",
    className: "g-icon"
  }), /*#__PURE__*/_react.default.createElement("figcaption", null, initial)) : null, type === 'avatar' ? /*#__PURE__*/_react.default.createElement("figure", null, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    icon: "g-tick",
    width: "24px",
    wrapperClassName: "g-chip-selected",
    className: "g-icon"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: avatar,
    alt: "chip-avatar"
  })) : null, /*#__PURE__*/_react.default.createElement("span", null, children), hasCloseIcon ? /*#__PURE__*/_react.default.createElement(_Icon.default, {
    icon: "g-cross",
    width: "16px",
    wrapperClassName: "g-chip-close",
    className: "g-icon"
  }) : null);
}

var _default = Chip;
exports.default = _default;
Chip.propTypes = {
  children: _propTypes.default.symbol,
  isEnabled: _propTypes.default.bool,
  isSelected: _propTypes.default.bool,
  isDisabled: _propTypes.default.bool,
  hasCloseIcon: _propTypes.default.bool,
  type: _propTypes.default.string,
  avatar: _propTypes.default.string,
  initial: _propTypes.default.string
};
Chip.defaultProps = {
  children: undefined,
  isEnabled: false,
  isSelected: false,
  isDisabled: false,
  hasCloseIcon: false,
  type: '',
  avatar: '',
  initial: ''
};
